# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/dns/DNS.h
# Commit: 96bd98319c62
# Full Hash: 96bd98319c6242a9b05dc4723b928e88537424e1
# Author: Kershaw Chang <kershaw@mozilla.com>
# Date: 2020-08-26 03:24:37
# Regressor Bug: 1652667
# File Overlap Count: 1
# Description:
#   Bug 1652667 - P1: Add telemetry for when an HTTPSSVC record is received r=dragana
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D87844
# ==============================================================================

diff -r b61a789151df -r 96bd98319c62 netwerk/dns/DNS.h
--- a/netwerk/dns/DNS.h	Tue Aug 25 17:29:21 2020 +0000
+++ b/netwerk/dns/DNS.h	Tue Aug 25 17:29:32 2020 +0000
@@ -46,6 +46,22 @@
 namespace mozilla {
 namespace net {
 
+// IMPORTANT: when adding new values, always add them to the end, otherwise
+// it will mess up telemetry.
+// Stage_0: Receive the record before the http transaction is created.
+// Stage_1: Receive the record after the http transaction is created and the
+//          transaction is not dispatched.
+// Stage_2: Receive the record after the http transaction is dispatched.
+enum HTTPSSVC_RECEIVED_STAGE : uint32_t {
+  HTTPSSVC_NOT_PRESENT = 0,
+  HTTPSSVC_WITH_IPHINT_RECEIVED_STAGE_0 = 1,
+  HTTPSSVC_WITHOUT_IPHINT_RECEIVED_STAGE_0 = 2,
+  HTTPSSVC_WITH_IPHINT_RECEIVED_STAGE_1 = 3,
+  HTTPSSVC_WITHOUT_IPHINT_RECEIVED_STAGE_1 = 4,
+  HTTPSSVC_WITH_IPHINT_RECEIVED_STAGE_2 = 5,
+  HTTPSSVC_WITHOUT_IPHINT_RECEIVED_STAGE_2 = 6,
+};
+
 // Required buffer size for text form of an IP address.
 // Includes space for null termination. We make our own contants
 // because we don't want higher-level code depending on things